net/local/local_sockif.c

```diff
@@ -718,6 +718,17 @@ static int local_ioctl(FAR struct socket *psock, int cmd,
 
   switch (cmd)
     {
+      case FIONBIO:
+        if (conn->lc_infile.f_inode != NULL)
+          {
+            ret = file_ioctl(&conn->lc_infile, cmd, arg);
+          }
+
+        if (conn->lc_outfile.f_inode != NULL)
+          {
+            ret = file_ioctl(&conn->lc_outfile, cmd, arg);
+          }
```

Comment:
```suggestion
        if (conn->lc_infile.f_inode != NULL)
          {
            ret = file_ioctl(&conn->lc_infile, cmd, arg);
          }
        else if (conn->lc_outfile.f_inode != NULL)
          {
            ret = file_ioctl(&conn->lc_outfile, cmd, arg);
          }
        else
          {
            ret = -ENOTCONN;
          }
```
I'm not sure if that is correct, but returning `OK` if both `conn->lc_infile.f_inode` and `conn->lc_outfile.f_inode` are NULL seems not correct to me.

Reply:
That's not right if write like that.

The local socket has two pipe connections, both of the need to be set.

I did some small change, please see the new version.

Reply:
Now it is much better. Thank you. By sample code I just wanted to highlight that overwriting of a failed `ret` is wrong and returning `OK` when both pipes are NULL is wrong as well.